LEAF_Request_Portal/sources/FormWorkflow.php

```diff
@@ -222,6 +222,97 @@ private function includePersonDesignatedData(array $srcRecords, array $pdRecords
         return $srcRecords;
     }
 
+    /**
+     * includeGroupDesignatedData efficiently merges approver data to $srcRecords, for a
+     * given list of $gdRecordList and $gdIndicators.
+     * 
+     * When $skipNames is false, $srcRecords is expected to contain dependencyIDs
+     * 
+     * WARNING: This function should only be used to support getRecordsDependencyData() and getActionable().
+     *          Usage in other areas must be carefully reviewed as this retrieves data without
+     *          checking for valid access. The $gdIndicators variable must only contain indicator IDs
+     *          that are related to a "group designated" field AND workflow that utilizes the "group 
+     *          designated" feature.
+     * @param array $srcRecords list of records
+     * @param array $gdRecordsMap map of record IDs => indicatorID that utilize "group designated"
+     * @param array $gdIndicator list of indicator IDs mapped to "group designated" fields
+     * @param bool $skipNames set true to exclude employee lookups
+     * @return array Amended records
+     */
+    private function includeGroupDesignatedData(array $srcRecords, array $gdRecordsMap, array $gdIndicators, bool $skipNames = false): array
+    {
+        // sanitize for use in query
+        $gdIndicators = array_map(function($x) {
+                            return (int)$x;
+                        }, $gdIndicators);
+        $gdIndicators = implode(',', $gdIndicators);
+
+        $gdRecordIDs = array_keys($gdRecordsMap);
+        $gdRecordIDs = array_map(function($x) {
+                        return (int)$x;
+                    }, $gdRecordIDs);
+
+        $gdRecordIDs = implode(',', $gdRecordIDs);
+
+        $query = "SELECT recordID, `data`, `name`, indicatorID FROM `data`

```

Comment:
Is there a reason you're not using the "standard" PDO practice of feeding the variables for the sql statement in via the variables array here and above?

Reply:
Yes, this was primarily to balance complexity and benefit, as parameterizing this would involves more logic. Even if the "WHERE IN" query was parameterized, the query would still have to be directly modified, necessitating validation of the modification, due to the unknown length of the IN(?, ?, ?,...) component. Since these queries only deal with integers, it's more straightforward to simply check the integers and make the validation easier to understand.